[Docs] Add Encoding Rules for Materials (#162)
raphaellith/Manuscripta: docs/Material Encoding.md

    (b) A separator row of dashes (`---`) shall follow the header row.
    (c) Column alignment shall be specified using colons in the separator row:
        (i) `:---` — Left-aligned.
        (ii) `:---:` — Centre-aligned.
        (iii) `---:` — Right-aligned.

(7) **LaTeX**. Mathematical notation shall be supported using LaTeX syntax:

    (a) Inline LaTeX shall be delimited by single dollar signs: `$...$`.
    (b) Block LaTeX shall be delimited by double dollar signs: `$$...$$`.

## Section 3: Attachment References

(1) **Image Syntax**. Images shall be embedded using standard markdown image syntax with attachment paths:

```
![alt text](/attachments/{id})
```

Where `{id}` shall be the UUID of the attachment.

(2) **Attachment Resolution**. Applications shall resolve attachment references by calling the `GET /attachments/{id}` endpoint as specified in API Contract §2.1.3.

(3) **Invalid References**. If an attachment reference cannot be resolved, applications should display a placeholder indicating the missing resource.

## Section 4: Custom Markers

(1) **General Syntax**. Custom markers shall use a fence-style syntax:

```
:::{marker-name}{attributes}
content
:::

> ```

Reviewer: ````suggestion
(1) **General Syntax**. Custom markers shall use the Admonition syntax as supported by the flexmark-java library:

```
!!! marker-name attribute="value"
    content
```
````

> Where `{marker-name}` shall be the marker type and `{attributes}` shall be a set of key-value pairs in the format `key="value"`.

Reviewer: ```suggestion
Where `marker-name` shall be the marker type and attributes shall be specified as space-separated key-value pairs. Content within the marker shall be indented by four spaces.
```

(2) **PDF Embed**. PDF documents shall be embedded using the `pdf` marker:

```
:::pdf{id="attachment-uuid"}

> :::

Reviewer: ```suggestion
!!! pdf id="attachment-uuid"
```

```

    (a) The `id` attribute shall be mandatory and shall reference a valid attachment.
    (b) Applications shall render an embedded PDF viewer or a download link.

(3) **Centred Text**. Text shall be centred using the `center` marker:

```
:::center
This text will be centred.
:::
```

    (a) Content within the marker may include other markdown elements.

>     (b) Nested custom markers shall not be supported.

Reviewer: ````suggestion
!!! center
    This text will be centred.
```

    (a) Content within the marker may include other markdown elements, each line indented by four spaces.
    (b) Nested custom markers shall not be supported.
````

(4) **Question Reference**. A reference to a `QuestionEntity` shall be embedded using the `question` marker:

```
:::question{id="question-uuid"}

> :::

Reviewer: ```suggestion
!!! question id="question-uuid"
```

    (b) Applications shall render the referenced question inline, including the question text and any options or input fields as appropriate to the `QuestionType`.
    (c) The referenced `QuestionEntity` shall have a `MaterialId` matching the containing material's ID. If the referenced `QuestionEntity` does not exist or is not associated with the material, applications shall display a placeholder indicating the missing resource.

## Appendix 1: Syntax Summary

| Element | Syntax | Example |
|---------|--------|---------|
| Header 1 | `# text` | `# Introduction` |
| Header 2 | `## text` | `## Chapter 1` |
| Header 3 | `### text` | `### Section 1.1` |
| Bold | `**text**` | `**important**` |
| Italic | `*text*` | `*emphasis*` |
| Unordered list | `- item` | `- First point` |
| Ordered list | `1. item` | `1. Step one` |
| Table | `\| col \| col \|` | See §2(6) |
| Inline LaTeX | `$...$` | `$x^2 + y^2 = z^2$` |
| Block LaTeX | `$$...$$` | `$$\int_0^1 f(x) dx$$` |
| Image | `![alt](/attachments/id)` | `![diagram](/attachments/abc-123)` |
| PDF embed | `:::pdf{id="..."}:::` | `:::pdf{id="abc-123"}:::` |
| Centred text | `:::center ... :::` | See §4(3) |

> | Question ref | `:::question{id="..."}:::` | See §4(4) |

Reviewer: ```suggestion
| PDF embed | `!!! pdf id="..."` | `!!! pdf id="abc-123"` |
| Centred text | `!!! center` | See §4(3) |
| Question ref | `!!! question id="..."` | See §4(4) |
```
